plugins/providers/policytag/config.go

```diff
@@ -0,0 +1,175 @@
+package policytag
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/mitchellh/mapstructure"
+	"github.com/odpf/guardian/domain"
+	"github.com/odpf/guardian/utils"
+)
+
+const (
+	AccountTypeUser           = "user"
+	AccountTypeServiceAccount = "serviceAccount"
+)
+
+// Credentials is the authentication configuration used by the policy_tag client
+type Credentials struct {
+	ServiceAccountKey string `mapstructure:"service_account_key" json:"service_account_key" validate:"required,base64"`
+	ResourceName      string `mapstructure:"resource_name" json:"resource_name" validate:"startswith=projects/"`
+}
+
+// Encrypt encrypts GCP credentials
+func (c *Credentials) Encrypt(encryptor domain.Encryptor) error {
+	if c == nil {
+		return ErrUnableToEncryptNilCredentials
+	}
+
+	encryptedCredentials, err := encryptor.Encrypt(c.ServiceAccountKey)
+	if err != nil {
+		return err
+	}
+
+	c.ServiceAccountKey = encryptedCredentials
+	return nil
+}
+
+// Decrypt decrypts GCP credentials
+func (c *Credentials) Decrypt(decryptor domain.Decryptor) error {
+	if c == nil {
+		return ErrUnableToDecryptNilCredentials
+	}
+
+	decryptedCredentials, err := decryptor.Decrypt(c.ServiceAccountKey)
+	if err != nil {
+		return err
+	}
+
+	c.ServiceAccountKey = decryptedCredentials
+	return nil
+}
+
+// Permission is for mapping role into policy tag fin-grained-reader permissions
```

Comment:
can you fix the typo in the comment here

Reply:
done